### PR TITLE
feat: Add upgradeable GHO Token

### PR DESCRIPTION
**Pull request by miguelmtzinf**
It might be preferable to use an upgradeable version of the GHO Token on chains other than Ethereum to allow for amendments if necessary.

This PR introduces an upgradeable version of the Gho Token contract. It retains the same functionality as the standard [GhoToken](https://github.com/aave/gho-core/blob/f02f87482de7ccbd30ba76b40939fb016dbb2fea/src/contracts/gho/GhoToken.sol) but includes minor adaptations to facilitate future upgrades. Notably, initializations typically performed in the constructor are now executed in the initialize function.

### CODE REVIEW

**Comment by miguelmtzinf** [comment]
Closes #398

**Comment by miguelmtzinf** [comment]
Closes #405

**Review by JoaquinBattilana — APPROVED** [review]

**Review by foodaka — APPROVED** [review]

File: src/contracts/gho/UpgradeableGhoToken.sol [hunk 1]
```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {EnumerableSet} from '@openzeppelin/contracts/utils/structs/EnumerableSet.sol';
+import {AccessControl} from '@openzeppelin/contracts/access/AccessControl.sol';
+import {VersionedInitializable} from '@aave/core-v3/contracts/protocol/libraries/aave-upgradeability/VersionedInitializable.sol';
```
[the remark this answers is not in the document]
Comment (sakulstra) [reply src/contracts/gho/UpgradeableGhoToken.sol:6]: same on ccip repo

[the remark this answers is not in the document]
Comment (miguelmtzinf) [reply src/contracts/gho/UpgradeableGhoToken.sol]: this has been resolved in [1576e9b](https://github.com/aave/gho-core/pull/400/commits/1576e9bd7f2cda7bcd9150714300d9d25de88c39)

**Comment by miguelmtzinf** [comment]
Diffs between GHO and UpgradeableGHO contracts can be found [here](https://github.com/aave/gho-core/pull/411)

**Comment by gitkit13** [comment]
> It might be preferable to use an upgradeable version of the GHO Token on chains other than Ethereum to allow for amendments if necessary.
> 
> This PR introduces an upgradeable version of the Gho Token contract. It retains the same functionality as the standard [GhoToken](https://github.com/aave/gho-core/blob/f02f87482de7ccbd30ba76b40939fb016dbb2fea/src/contracts/gho/GhoToken.sol) but includes minor adaptations to facilitate future upgrades. Notably, initializations typically performed in the constructor are now executed in the initialize function.

**Pull request merged**